async-nats/src/lib.rs

```diff
@@ -585,21 +519,115 @@ impl SubscriptionContext {
     }
 }
 
-/// A connector which facilitates communication from channels to a single shared connection.
-/// The connector takes ownership of the channel.
+///
```

Comment:
Let's add some basic docs here at lesat.

Reply:
@caspervonb  lets fix this :)

Reply:
Split Connector out into https://github.com/nats-io/nats.rs/pull/392